File: Kikaria-Android/app/src/main/java/com/vita0818/kikaria/util/LatexToken.kt
package com.vita0818.kikaria.util

/**
 * LaTeX token types, translated from LatexToken.swift.
 *
 * Represents parsed elements from a knowledge point content string
 * that contains LaTeX math markup ($...$ and $$...$$).
 */
sealed class LatexToken {
    data class Text(val value: String) : LatexToken()
    data class InlineMath(val source: String, val body: String) : LatexToken()
    data class BlockMath(val source: String, val body: String) : LatexToken()
    data class Fallback(val value: String) : LatexToken()
}

File: Kikaria-LegacyiOS/Kikaria/LatexToken.swift
//
//  LatexToken.swift
//  Kikaria
//
//  Created by Codex on 2026/5/10.
//

import Foundation

enum KikariaLatexToken: Equatable {
    case text(String)
    case inlineMath(source: String, body: String)
    case blockMath(source: String, body: String)
    case fallback(String)
}
